src/main/java/org/spongepowered/api/world/generation/GenerationRegion.java

```diff
@@ -51,8 +51,13 @@
  * nor is it considered to be viewable by {@link Player}s or accessible
  * by the {@link Server} or {@link World}(s).
  */
-public interface GenerationRegion extends ProtoWorld<GenerationRegion> {
+public interface GenerationRegion extends WorldLike<GenerationRegion> {
 
+    /**
+     * The {@link ServerWorld} that this region will be placed into.
+     *
+     * @return The world
+     */
     ServerWorld world();
```

Comment:
Maybe add a note that the world may not be safe to access asynchronously still? (Even Mojang locks this down a bit internally)

Reply:
I think the better option here is to just provide the world key to try to discourage that further.